### DISCUSSION TITLE
Cancelling runs in Cypress Cloud programmatically

### DISCUSSION THREAD

**Discussion by cgutierrezpa** [cypress-io/cypress#27051] [Cypress Cloud]
Hi all,

Did anyone come across a use case where they need to cancel a Cypress run in the Cloud dashboard from their CI in a programmatic way? I have the feeling that these Cloud functionalities are missing an API management feature, as we cannot handle anything within the Cloud dashboards from within the CI

For example, there are times in which some machines in the CI would "die" and timeout while running tests with `--parallel`. The problem is that Cypress Cloud is in charge of syncing with these machines and updating the status of the run. However, it never does, and the run can keep in a `running` state for hours. This gets worse when you integrate the Cloud GitHub status checks in the GitHub merge queue because it can literally starve the merge queue for hours, because GitHub never receives the status check update from Cypress Cloud side, and this is far from ideal. It would be very easy to simply handle these exceptions by sending a `CANCEL` signal to run in Cypress Cloud, and the status check would be synced immediately. 

Does anyone know how can we achieve this?

**Comment by tinu-doit-intl** [comment]
@jennifer-shehane any thoughts on this? Thank you.
